notes.md

```diff
@@ -358,6 +358,96 @@ it as 0x53B9 instead of 0x53B8. `Matroska Readers` may support these legacy file
+and the last contains karaoke versions of the music that plays during the film, which is an unusual
+specialized audio service that Matroska has no built-in support for indicating, so it's indicated
+in the track name instead. By not setting the Default flag on these specialized tracks, the file's author
+hints that they should not be automatically selected by a default-configured player.
+
+Having narrowed its choices down, our example player now may have to select between tracks 2 and 3.
+The only difference between these tracks is their channel layouts: 2 is 5.1 surround, while 3 is stereo.
+If the player is aware that the output device is a pair of headphones or stereo speakers, it may wish
+to prefer the stereo mix automatically. On the other hand, if it knows that the device is a surround system,
+it may wish to prefer the surround mix.
+
+If the player finishes analyzing all of the available audio tracks and finds that multiple seem equally
+and maximally preferable, it **SHOULD** default to the first of the group.
+
+## Subtitle selection
+
+Example track set:
+
+1. Video, lang=und
+2. Audio, lang=fra, Original, Default
+3. Audio, lang=por, Default
+4. Subtitles, lang=fra, Original, Default
+5. Subtitles, lang=fra, Original, Hearing-impaired
+6. Subtitles, lang=por, Default
+7. Subtitles, lang=por, Forced
+8. Subtitles, lang=por, Hearing-impaired
+
+Here we have 2 audio tracks and 5 subtitle tracks. As we can see, French is the original language.
+
+We'll start by discussing the case where the user prefers French (or Original-language)
+audio (or has explicitly selected the French audio track), and also prefers French subtitles.
+
+In this case, if the player isn't configured to display captions when the audio matches their
+preferred subtitle languages, the player doesn't need to select a subtitle track at all.
+
+If the user _has_ indicated that they want captions to be displayed, the selection simply
+comes down to whether Hearing-impaired subtitles are preferred.
+
+The situation for a user who prefers Portuguese subtitles starts out somewhat analogous.
+If they select the original French audio (either by explicit audio language preference,
+preference for Original-language tracks, or by explicitly selecting that track), then the
+selection once again comes down to the hearing-impaired preference.
+
+However, the case where the Portuguese audio track is selected has an important catch:
+a Forced track in Portuguese is present. This may contain translations of onscreen text
+from the video track, or of portions of the audio that are not translated (music, for instance).
+This means that even if the user's preferences wouldn't normally call for captions here,
+the Forced track should be selected nonetheless, rather than selecting no track at all.
+On the other hand, if the user's preferences _do_ call for captions, the non-Forced tracks
+should be preferred, as the Forced track will not contain captioning for the dialogue.
```

Comment:
Interresting. I think on DVDs the forced track doesn't have any language and is displayed no matter what. In this case the forced subtitles should have `lang=und` and be used no matter what.

Maybe we need another example with forced tracks where you have multiple language variants and an `und` fallback version which should be used even if there's no subtitle selected. IMO the forced flag is the most tricky one.

Reply:
> I think on DVDs the forced track doesn't have any language and is displayed no matter what.

DVDs don't generally have distinct forced tracks at all (though theoretically an author can set up the menus and media content any way they like); instead, they generally have a single track for each language, with individual events flagged as either forced or not at the codec level; same's true for blu-ray. Some software for converting DVD and BD media to Matroska has an option to create 2 tracks (one regular containing all events, and one containing only the forced ones, the latter having the matroska forced flag set). As far as I'm aware, this "per-event forced" concept doesn't exist in any subtitle codecs other than the DVD and BD ones.

> Maybe we need another example with forced tracks where you have multiple language variants and an `und` fallback version which should be used even if there's no subtitle selected.

What for? I've never heard of anyone trying to use the flag this way, and I can't think of a use-case (why would you want to display an additional text overlay that isn't part of the video, and that isn't in a language the user can read?). Even if we did say "if a forced `und` track exists and no other forced track matches the user's language preferences, default to it", saying "even if there's no subtitle selected" implies that the user can't disable subtitles altogether, which I can basically guarantee that no player is going to comply with.

Reply:
I don't understand the remark of `saying "even if there's no subtitle selected" implies that the user can't disable subtitles altogether, which I can basically guarantee that no player is going to comply with`. I don't think it implies that. You can decide for a file/stream you don't want subtitles. That doesn't mean your preferred languages are not stored (they should be in your OS at the very least).

Multiple forced tracks should be cover. It may exist in the real world, and if it doesn't it should. For example imagine a scene where the whole dialog is not verbal but via SMS, displayed as forced subtitles over the video (I just had the example yesterday in Euphoria). This is a forced track, because even if you don't want subtitles (my case) you still want to know what they are saying (part of the plot). But I'd rather have it in English (also the original language) than French (what I actually got). So at least this stream should have 2 "forced" tracks. And the player should pick one for the user (English or original in my case).

So yes, maybe the `lang=und` doesn't make a lot of sense here. It might be used for imaginary languages (elf) which doesn't really have an ISO value.

In the end the language selection of the subtitle forced track is more or less the same as the regular subtitle track. It might be worth mentioning that there should only be one of each at any time.

Reply:
> I don't understand the remark of `saying "even if there's no subtitle selected" implies that the user can't disable subtitles altogether, which I can basically guarantee that no player is going to comply with`. I don't think it implies that. You can decide for a file/stream you don't want subtitles. That doesn't mean your preferred languages are not stored (they should be in your OS at the very least).

I don't quite get what you were trying to say to begin with then.

> Multiple forced tracks should be cover.

This is already handled, it's just not shown in the example currently.

> It may exist in the real world, and if it doesn't it should.

It definitely does. Take e.g. the Godfather trilogy: the blu-rays have regular captioning in several languages, but the tracks corresponding to the languages audio is available in (at a minimum, english, german, and french) also have forced events showing translations of dialogue spoken in Italian, and some onscreen text (like overlays naming the location or time period of a scene). So in Matroska, you end up with several regular tracks and 3 forced ones, and track selection (under the rules I'm proposing, at least) works exactly the way you're asking for it to.

> It might be used for imaginary languages (elf) which doesn't really have an ISO value.

I've never seen a subtitle track _in_ Elvish or some other fictional language, but the standard ISO code for that would be `mis` ("uncoded languages").

> In the end the language selection of the subtitle forced track is more or less the same as the regular subtitle track.

Basically, yeah; the language part is identical for either case, and then you just pick either a forced track or a non-forced one (or nothing if there isn't a track in the target language of the type you're looking for).

Reply:
So my point was just the forced track selection should have an example of its own. In particular because it exists in parallel of the regular subtitles and may have different choices of languages. But it can be done later.